shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -349,8 +349,15 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
     }
 
     pragma[nomagic]
+    private NodeRegion getNodeRegion(NodeEx n) { result.contains(n.asNode()) }
+
+    bindingset[n, cc]
+    pragma[inline_late]
```

Comment:
I'm a bit unsure whether we actually want to make this change as-is. There's a lot of subtlety here, since this change builds in further assumptions about `NodeRegion` disjointness. What if, instead of the changes on this PR, we simply changed `nomagic` to `inline_late` on the existing `isUnreachableInCall1`? That would make the materialisation cached.

Reply:
That also works for me; I have opened a new PR.